### PR TITLE
core.async was having trouble compiling this case expr

### PR DESCRIPTION
**Pull request by ctdean**
Rewrite to use condp instead.

### CODE REVIEW

**Comment by ctdean** [comment]
@jimbru please review

File: src/backtick/engine.clj [hunk 1]
```diff
@@ -78,22 +78,23 @@
...
-                 (let [[done? port] (alts! [done-ch (timeout (:timeout-ms master-cf))])]
-                   (if done?
-                       (db/queue-finish! (select-keys msg [:id]))
-                       (do
-                         (when worker
-                           (.cancel worker true))
-                         (log/infof "job %s timeout: %s" job (:name msg)))))
-                 (log/debugf "start-jobs: waiting")
-                 (recur)))))))
+                           worker (submit-worker pool done-ch msg)]
```
Comment (jimbru) [inline src/backtick/engine.clj:87]: Your tabbing in this section looks messed up?

**Comment by jimbru** [comment]
Fcking core async compiler. :shipit:

File: src/backtick/engine.clj (hunk 1 above)
Comment (ctdean) [inline src/backtick/engine.clj:87]: Yep, let me fix that upstream

**Pull request merged**